### PR TITLE
Bugfix: improve layout of Preferences > Shortcuts tables

### LINKED ISSUE #5671: Shortcuts preferences: long action names are elided even though there is space
## 🐛 Bug

Long action names, like `M` in the below screenshot are elided and it's not clear what the keybind does:
image
(it's `Set the currently selected label to the largest used label plus one.`)

Note that no amount of resizing of the window affects this. It's like 1/3 1/3 1/3 split between the action name and the two keybindings.
Note: it's possible to resize the action description column, but it's also not helpful:
image

The mouse-over tooltip is also totally unhelpful `Set keyboard shortcuts for action`.

## To Reproduce

1. Open the Preferences / Shortcuts panel
2. Select `Labels` from the dropdown

## Expected behavior

The action name should line wrap? or there should be less room for keybinds? Or helpful tooltip.

## Environment

```

napari: 0.5.0a2.dev51+gfc038f36
Platform: macOS-13.2.1-arm64-arm-64bit
System: MacOS 13.2.1
Python: 3.10.9 | packaged by conda-forge | (main, Feb 2 2023, 20:26:08) [Clang 14.0.6 ]
Qt: 5.15.6
PyQt5: 5.15.7
NumPy: 1.24.2
SciPy: 1.10.1
Dask: 2023.3.1
VisPy: 0.12.2.dev45
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.2
npe2: 0.6.2.dev26+gd7329b7.d20230324

OpenGL:
- GL version: 2.1 Metal - 83
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Settings path:
- /Users/piotrsobolewski/Library/Application Support/napari/napari-dev_a1eb8b76ba95fa16ad06e26097b46b8455dfbf0b/settings.yaml
```

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# Description
Closes https://github.com/napari/napari/issues/5671

This PR adjusts the layout of the tables used for editing shortcuts such that long action descriptions can be read in their entirety. Briefly:
1) it changes the column widths, to give the text more room
2) as a result of 1, it left aligns the action descriptions so they are visually closer to the keybindings
3) it enables wrapping of long action descriptions
4) as a result of 3, it adjusts the row height and padding

Here's the new look: Viewer
image

versus `main` without this PR:
image

Notice how `Show/hide IPython ...` action has the full description readable, because it's line wrapped. Also, in my opinion the overall layout is more readable and pleasing.
Open to suggestions though!

New look Labels (main is shown in the issue):
image

Note: the new widths of the keybind columns are set to be sufficient for Ctrl-Alt-Meta-<Letter>  with a bit extra left-over.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Closes https://github.com/napari/napari/issues/5671
Some discussion of the table is also here https://github.com/napari/napari/pull/5018
this was the PR that implemented the second keybinding column.

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Carreau** [comment]
That will likely conflict with #5604 and #5064, but +1

**Comment by Czaki** [comment]
As #5696 is merged, could you update the branch and post new screenshots?

**Comment by psobolewskiPhD** [comment]
> As #5696 is merged, could you update the branch and post new screenshots?

Yes, sorry was traveling. I will get on this this weekend.

**Comment by psobolewskiPhD** [comment]
I think it will be impossible to satisfy pyside6, pyqt6 and pyqt5 all the way back to 5.12.3
I think pyqt5 5.13 is needed for the alignment flags to work.
So either need some logic to check Qt version or bump minimum requirements...

**Comment by Czaki** [comment]
Maybe add `napari._qt._compat` and import the alignment flag from `_compat`?

But also 0.5.0 may be enough to bump Qt requirements.

**Comment by Czaki** [comment]
Could you update the screenshots in the description?

**Comment by psobolewskiPhD** [comment]
> Could you update the screenshots in the description?

Done. Sorry took so long.

Question remains regarding the min-req. Could try bumping to 5.13 and see if tests pass?

**Comment by Czaki** [comment]
For me you could bump. 

Looking on the screenshot (and locally on napari), there is a problem with the readability of Action description when using a dark theme (as text is to dark). Did you agree with my observation?

**Comment by psobolewskiPhD** [comment]
> For me you could bump.

I see that pyside2 is already min >5.13 so making pyqt5 also >5.13 makes some sense there.

> Looking on the screenshot (and locally on napari), there is a problem with the readability of Action description when using a dark theme (as text is to dark). Did you agree with my observation?

Yes, I see it now. It's not related to this PR, because it happens on main. I will debug and I think it's better to make a separate bugfix PR.

**Comment by psobolewskiPhD** [comment]
!!
Reading up on flags and stuff for https://github.com/napari/napari/pull/5854 I realized how to fix the issue with the enums and flags here!
It's working in qt6 and qt5, without messing with min req
🎉

File: napari/_qt/widgets/qt_keyboard_settings.py [hunk 1]
```diff
@@ -232,6 +236,9 @@ def _set_table(self, layer_str: str = ''):
                 item = QTableWidgetItem(action.description)
                 item.setFlags(Qt.ItemFlag.NoItemFlags)
                 self._table.setItem(row, self._action_name_col, item)
+                # Ensure long descriptions can be wrapped in cells
+                item.textAlignment = Qt.AlignRight | Qt.AlignVCenter
```
Comment (Czaki) [inline napari/_qt/widgets/qt_keyboard_settings.py]:
```suggestion
                item.textAlignment = Qt.AlignmentFlag.AlignRight | Qt.AlignmentFlag.AlignVCenter
```

Why not use an enum explicitly?

Comment (psobolewskiPhD) [reply napari/_qt/widgets/qt_keyboard_settings.py]: I think that broke with the old qt. But yea using the Enum is better for Qt6 long term?
Let's try.

Comment (Czaki) [reply napari/_qt/widgets/qt_keyboard_settings.py]: Yes.

**Review by Czaki — APPROVED** [review]
As https://github.com/napari/napari/pull/5854 is merged only update screenshot (to not grayed action description left)

**Comment by psobolewskiPhD** [comment]
Coming back to this, I sort of hated the right justify after all.
I ended up using non-breaking spaces in the Viewer action names and changing the columns to be a bit more balanced.
Then left justify everything (including headers)

(I updated the screenshots in OP)

**Comment by jni** [comment]
@psobolewskiPhD I might call this an enhancement more than a bugfix?

**Comment by psobolewskiPhD** [comment]
@jni I guess? depends on how you look at the truncated action names.
🤷‍♂️ 
But sure, we can call it enhancement.

**Comment by Czaki** [comment]
Why move to 0.5.0? It is not breaking change, so it is worth trying to backport it.

**Comment by psobolewskiPhD** [comment]
I was going off @jni comment, thought he was meaning to move it to 0.5.0 because it's not a bug fix?

**Pull request merged**